### PR TITLE
blog: add dotnet component blog

### PR DESCRIPTION
**Pull request by ericgregory**
~~WIP: draft complete and ready for review, publish once wash 0.32.1 is released~~

wash 0.32.1 is out, so this is ready to go

question for @lachieh : How do you feel about using webp for the blog header? The smaller size is appealing, but wanted to check

### CODE REVIEW

File: blog/2024-09-05-build-wasm-components-dotnet-wasmcloud/index.mdx [hunk 1]
```diff
@@ -0,0 +1,384 @@
...
+
+For this blog, we'll use the [.NET 9 Preview 7 SDK](https://dotnet.microsoft.com/en-us/download/dotnet/9.0). If you're a Visual Studio Code user, you may also want to [install the C# Dev Kit extension](https://marketplace.visualstudio.com/items?itemName=ms-dotnettools.csdevkit), but in this walkthrough, we'll stick to the command line. 
+
+:::warning[Windows-only, for now]
+At the moment, `componentize-dotnet` only works on Windows. Maintainers expect macOS and Linux support to land soon.
+:::
+
+In addition to the .NET 9 SDK, you'll need:
+
+* [`wasmtime`](https://wasmtime.dev/): This is the standalone WebAssembly runtime that we'll use to test some components locally
```
Comment (CazJT) [inline blog/2024-09-05-build-wasm-components-dotnet-wasmcloud/index.mdx]: full-stop at the end of this sentence

File: blog/2024-09-05-build-wasm-components-dotnet-wasmcloud/index.mdx [hunk 2]
```diff
@@ -0,0 +1,384 @@
...
+        [DllImport("wasi:http/types@0.2.0", EntryPoint = "[resource-drop]outgoing-response"), WasmImportLinkage]
+        private static extern void wasmImportResourceDrop(int p0);
+...
+```
+
+In this file we see the types for `wasi:http` bound to C# with documentation from the original WIT included throughout the binding file. 
+
+If we look back at `Program.cs`, we see some of those types in use, including `OutgoingResponse` and `ResponseOutparam`. Consulting the binding files, we can make use of any WIT interface in our code.
+
+All right&mdash;we've already built our component. Let's test it out.
```
Comment (CazJT) [inline blog/2024-09-05-build-wasm-components-dotnet-wasmcloud/index.mdx:261]: Alright?

**Review by CazJT — COMMENTED** [review]
Small comments

File: blog/2024-09-05-build-wasm-components-dotnet-wasmcloud/index.mdx [hunk 3]
```diff
@@ -0,0 +1,384 @@
...
+
+Since this component uses HTTP, we'll use the `wasmtime serve` subcommand to run the component at `localhost:3000`:
+
+```sh
```
Comment (brooksmtownsend) [inline blog/2024-09-05-build-wasm-components-dotnet-wasmcloud/index.mdx]: I'd suggest using "```powershell" for all of the triple backtick shell blocks that we're running on Windows, it gives better syntax highlighting 😄

File: blog/2024-09-05-build-wasm-components-dotnet-wasmcloud/index.mdx [hunk 4]
```diff
@@ -0,0 +1,384 @@
...
+```
+```text
+Hello, World!
+```
+You can CTRL+C to stop the `serve`. When we inspect the component with `wash` this time, we'll see that our component has a different export:
+
+```sh
+wash inspect --wit bin\Debug\net9.0\wasi-wasm\native\hello-wasm.wasm
+```
+```text
```
Comment (brooksmtownsend) [inline blog/2024-09-05-build-wasm-components-dotnet-wasmcloud/index.mdx]:
```suggestion
```wit
```

I'm fairly sure we support this highlight

File: blog/2024-09-05-build-wasm-components-dotnet-wasmcloud/index.mdx [hunk 5]
```diff
@@ -0,0 +1,384 @@
...
+Hello, World!
+The current time is 09/03/2024 19:49:03
+```
+
+We can also use `wash inspect` to view the WASI interfaces used by this component:
+
+```sh
+wash inspect --wit bin\Debug\net9.0\wasi-wasm\native\hello-wasm.wasm
+```
+```text
```
Comment (brooksmtownsend) [inline blog/2024-09-05-build-wasm-components-dotnet-wasmcloud/index.mdx]:
```suggestion
```wit
```

I'm fairly sure we support this highlight

File: blog/2024-09-05-build-wasm-components-dotnet-wasmcloud/index.mdx [hunk 6]
```diff
@@ -0,0 +1,384 @@
...
+wasmtime bin\Debug\net9.0\wasi-wasm\native\hello-wasm.wasm
+```
+```text
+Hello, World!
+The current time is 09/03/2024 19:49:03
+```
+
+We can also use `wash inspect` to view the WASI interfaces used by this component:
+
+```sh
```
Comment (brooksmtownsend) [inline blog/2024-09-05-build-wasm-components-dotnet-wasmcloud/index.mdx]:
```suggestion
```powershell
```

for example

**Review by brooksmtownsend — COMMENTED** [review]
looks nice! I have all requests for changing the syntax highlighting inside of the triple backtick blocks, so leaving that as a comment

File: blog/2024-09-05-build-wasm-components-dotnet-wasmcloud/index.mdx (hunk 3 above)
Comment (ericgregory) [reply blog/2024-09-05-build-wasm-components-dotnet-wasmcloud/index.mdx]: Nice to know! Thanks, good shout

**Comment by ericgregory** [comment]
> looks nice! I have all requests for changing the syntax highlighting inside of the triple backtick blocks, so leaving that as a comment

Thanks, updated!

**Comment by lachieh** [comment]
> question for @lachieh : How do you feel about using webp for the blog header? The smaller size is appealing, but wanted to check

Love it. The more size we can shed the better. Webp is baseline supported for all browsers, so go ham!

**Review by brooksmtownsend — APPROVED** [review]

**Pull request merged**